### PR TITLE
 Add  ill-conditioned training check method for kriging-based surrogates, add SGP notebook

### PR DESCRIPTION
**Pull request by Paul-Saves**
- Fix https://github.com/SMTorg/smt/issues/585
- Add sgp analytical examples notebook
- Fix a typo in doc of mfk
- MGP was not tested properly within test_all.py

### CODE REVIEW

File: smt/surrogate_models/krg_based.py [hunk 1]
```diff
@@ -520,6 +520,24 @@ def _new_train(self):
...
+        print(
+            "Minimal distance between two points in any dimension is",
+            "{:.2e}".format(np.min(self.D)),
+        )
+        print(
+            "Correlation matrix R condition number is",
+            "{:.2e}".format(
+                np.linalg.cond(self.optimal_par["C"] @ self.optimal_par["C"])
+            ),
+        )
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:539]: If we could agree on some threshold values we could return a boolean indicating potential well or ill condition (maybe the function could be renamed `is_training_ill_conditioned() ?`). We could also use the `self.printer` mechanism to be able to enable/disable messages.

**Review by relf — CHANGES_REQUESTED** [review]
The PR should contain only the changes related to the new function check of ill-conditionned training.

File: smt/applications/tests/test_mfk.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def test_mfk_derivs(self):
         for kx in range(prob.xlimits.shape[0]):
             dye[kx] = prob(xe, kx=kx)
 
-        sm = MFK(theta0=[1e-2] * self.ndim)
+        sm = MFK(theta0=[1e-2] * self.ndim, corr="squar_exp")
```
[the remark this answers is not in the document]
Comment (Paul-Saves) [reply smt/applications/tests/test_mfk.py:115]: I think it is better to precise it

**Review by relf — CHANGES_REQUESTED** [review]
I think we do not need all the test tolerance changes as there is no change in the algorithms. Please could you edit the title and description of the PR to focus on the new `is_training_ill_conditioned` function for kriging-based surrogates? I would have preferred the notebook addition in another dedicated PR.

**Pull request merged**